src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -33,13 +44,16 @@ public JsonRpcSocketsClient(
         IJsonRpcLocalStats jsonRpcLocalStats,
         IJsonSerializer jsonSerializer,
         JsonRpcUrl? url = null,
-        long? maxBatchResponseBodySize = null)
+        long? maxBatchResponseBodySize = null,
+        int concurrency = 1)
         : base(clientName, stream, jsonSerializer)
     {
         _jsonRpcProcessor = jsonRpcProcessor;
         _jsonRpcLocalStats = jsonRpcLocalStats;
         _maxBatchResponseBodySize = maxBatchResponseBodySize;
         _jsonRpcContext = new JsonRpcContext(endpointType, this, url);
+        _processChannel = Channel.CreateBounded<ProcessRequest>(concurrency);
```

Comment:
Is it single writer, multiple reader? Let's provide that context for optimization.

Reply:
If I'm not mistaken, it's actually single reader AND single writer, but multiple workers.

Reply:
single writer, multiple reader.